core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -86,15 +89,12 @@
 
     private final StartStopSupport<Void, ServerListener> startStop;
     private final Set<Channel> serverChannels = Collections.newSetFromMap(new ConcurrentHashMap<>());
-    private final Map<InetSocketAddress, ServerPort> activePorts = new ConcurrentHashMap<>();
+    private final Map<InetSocketAddress, ServerPort> activePorts = new LinkedHashMap<>();
```

Comment:
This is not thread-safe because `activePorts` can be read from more than one thread via `activePort()` while `Server` mutates it.

Reply:
I missed what you mentioned. Just focused on adding and clearing that are not performing concurrently. :-( Thanks!